libexec/bootstrap-scripts/main-deffile.sh

```diff
@@ -40,6 +40,24 @@ if [ -z "${SINGULARITY_ROOTFS:-}" ]; then
     exit 1
 fi
 
+## For deffile bootstraps, we can implement a "minsize" guard.  This triggers a
+## size check against the image we're trying to bootstrap into, so we don't
+## waste time building into an image file that will be too small.
+#this environment variable might not be defined, so we can make it null.
+MINSIZE=${MINSIZE:-}
+if [ -n "$MINSIZE" ]; then
+  #report that it's been included in the def file.
+  echo "Minimum image size has been set: $MINSIZE M"
+  #get the size of the image, in megabytes.
+  IMAGE_SIZE=$(du --apparent-size -m $SINGULARITY_IMAGE | cut -f 1)
+  echo "Detected size of the image: $IMAGE_SIZE"
```

Comment:
@ityonemo could you comment on the choice to use --apparent-size? Just curious, I've never used it before. Can you confirm it has support for all old, crusty operating systems?